monai/data/dataset.py

```diff
@@ -580,7 +580,7 @@ def _transform(self, index: int):
         for _transform in self.transform.transforms:
             if start_run or isinstance(_transform, Randomizable) or not isinstance(_transform, Transform):
                 start_run = True
-                data = apply_transform(_transform, data)
+                data = apply_transform(_transform, deepcopy(data))
```

Comment:
I feel the deepcopy should happen after line 577？

Reply:
I would suggest below code to only deepcopy once:
```py
for _transform in self.transform.transforms:
    if start_run or isinstance(_transform, Randomizable) or not isinstance(_transform, Transform):
        if not start_run:
            start_run = True
            data = deepcopy(data)
        data = apply_transform(_transform, data)
```
@rijobro @wyli , what do you think?

Thanks.

Reply:
@Nic-Ma good suggestion, I did something similar using `enumerate` so we can delete `start_run`. I checked that the unit test passes with the modified code, and fails when I comment it out, so the deep copy is still working as intended.

Reply:
Hi @rijobro , the `start_run` var is used to mark the **first** random transform, so we only need to deepcopy once, then run all the random transforms based on the copied data.

Thanks.

Reply:
ah I understand, sorry about that. reverted back to using `start_run`.